bin/glm

```diff
@@ -0,0 +1,253 @@
+          formattedPublishedLabels,
+          localConfig
+        );
+        const labelsToUpdate = await checkForModifications(
+          localConfig,
+          formattedPublishedLabels
+        );
+        if (newLabels.length !== 0) {
+          await createLabels(newLabels);
+        }
+        if (labelsToDelete.length !== 0) {
+          await deleteLabels(labelsToDelete);
+        }
+        if (labelsToUpdate.length !== 0) {
+          await updateLabels(labelsToUpdate);
+        }
+        const changes =
+          newLabels.length + labelsToDelete.length + labelsToUpdate.length;
+        const summary =
+          changes !== 0
+            ? {
+                created: newLabels.length,
+                deleted: labelsToDelete.length,
+                updated: labelsToUpdate.length,
+              }
+            : 'Labels already up to date!';
+        resolve(summary);
+      });
+    } catch (err) {
+      reject(err);
+    }
+  });
+}
+
+function checkForNewOrOldLabels(newLabels, oldLabels) {
+  return new Promise((resolve, reject) => {
+    try {
+      const updatedLabels = [];
+      const OldLabelIds = [];
+      oldLabels.forEach(label => {
+        OldLabelIds.push(label.id ? label.id : undefined);
+      });
+      newLabels.forEach(label => {
+        if (!OldLabelIds.includes(label.id ? label.id : undefined)) {
+          updatedLabels.push(label);
+        }
+      });
+      resolve(updatedLabels);
+    } catch (err) {
+      reject(err);
+    }
+  });
+}
+
+function checkForModifications(localConfig, publishedConfig) {
+  return new Promise(async (resolve, reject) => {
+    const formattedPublishedConfig = await labelFormatter(publishedConfig);
+    const labelChanges = [];
+    try {
+      localConfig.forEach(localLabel => {
+        formattedPublishedConfig.forEach(publicLabel => {
+          if (localLabel.id === publicLabel.id) {
+            if (
+              Object.values(localLabel).toString() !==
+              Object.values(publicLabel).toString()
+            ) {
+              labelChanges.push(localLabel);
+            }
+          }
+        });
+      });
+      resolve(labelChanges);
+    } catch (err) {
+      reject(err);
+    }
+  });
+}
+
+function labelFormatter(labelData) {
+  return new Promise((resolve, reject) => {
+    try {
+      const formattedLabels = [];
+      labelData.forEach(label => {
+        const data = {
+          id: label.id,
+          name: label.name,
+          color: label.color,
+          description: label.description,
+        };
+        formattedLabels.push(data);
+      });
+      resolve(formattedLabels);
+    } catch (err) {
+      reject(err);
+    }
+  });
+}
+
+function fetchLabels() {
+  const url = `https://api.github.com/repos/${owner}/${repo}/labels`;
+  return axios.get(url).then(response => {
+    const labels = [];
+    response.data.forEach(label => {
+      labels.push(label);
+    });
```

Comment:
What is this for? You're reconstructing the same array as it was 🤔

Reply:
😅   567f587